Laboratory/Tests/L2-Semantics/_2_TypedFunctions.ele

```diff
@@ -17,4 +17,6 @@ intrinsic function add(a:Num, b:Num):Num
 addFive(a:Num):Num = add(a, 5)
 
 literalNumNotAConstraint(a:5):10 = a
-literalFunctionNotAConstraint(a:onlyNum):returnsNum = a
\ No newline at end of file
+literalFunctionNotAConstraint(a:onlyNum):returnsNum = a
+
+constraintLookupError(a:ConstraintThatDoesntExist) = a
```

Comment:
should this have a return constraint so it's a boundary function for runtime tests?

Reply:
While we need to add runtime tests for semantics stage - this particular one can't be runtime tested because the boundary constraints can't be serializable types. There's also no difference between compile-time and runtime in a case where the error should be reported at compile time.

Reply:
I think this is a difference between the two compilers then, in libelement I'm fairly sure it will first check if it's a boundary function and give an error about that before it tries to do constraint checking.

Reply:
This doesn't need to be checked as a boundary function because it's a compile-time error to have an unresolved constraint. It's not possible to determine if a function is boundary-compatible until the constraints are all evaluated.

Reply:
it's possible just by checking that there is a type annotation, but that's fair, we can leave it for now :)